client/state/site-settings/exporter/actions.js

```diff
@@ -26,41 +43,168 @@ export function setPostType( postType ) {
+
+		wpcomUndocumented.getExportSettings( siteId, ( error, data ) => {
+			dispatch( receiveExportSettings( siteId, data ) );
+		} );
+	}
+}
+
+/**
+ * Called when the available export settings are returned from the server.
+ *
+ * @param  {int}      siteId  The ID of the site for which the export settings belong
+ * @param  {Object}   data    The data returned from the server
+ * @return {Function}         Action thunk
+ */
+export function receiveExportSettings( siteId, data ) {
+	return {
+		type: RECEIVE_EXPORTER_ADVANCED_SETTINGS,
+		siteId: siteId,
+		data: data
+	};
+}
+
+/**
+ * Sends a request to the server to start an export.
+ *
+ * @param {string}      postType        The type of post to export, or undefined
+ * @return {Function}                   Action thunk
+ */
+export function startExport( postType ) {
+	// Setting startExport on an event handler passes the event object to
+	// the action creator's first parameter, causing confusing errors. This
+	// assertion will help avoid making this mistake.
+	if ( postType && typeof postType !== 'string' ) {
+		throw new Error(
+			'Expected postType parameter of startExport action creator to ' +
+			'be a string, got ' + ( typeof postType ) + '. Did you pass the action ' +
+			'creator directly to an event handler? Wrap it to make first param blank: ' +
+			'onClick={ ( event ) => startExport() }'
+		);
+	}
+
+	return ( dispatch, getState ) => {
+		const state = getState();
+		const advancedSettings = prepareExportRequest( state, postType );
+		const site = getSelectedSite( state );
+
+		if ( ! site ) {
+			return;
+		}
+
+		dispatch( {
+			type: REQUEST_START_EXPORT,
+			siteId: site.ID,
+			advancedSettings
 		} );
 
-		// This will be replaced with an API call to start the export
-		setTimeout( () => {
+		wpcomUndocumented.startExport( site.ID, advancedSettings, ( startError, startResponse ) => {
+			debug( startError, startResponse );
+			if ( startError ) {
+				dispatch( failExport( ERROR_START_EXPORT ) );
+				return;
+			}
+
 			dispatch( replyStartExport() );
 
-			// This will be replaced with polling to check when the export completes
-			setTimeout( () => {
-				dispatch( completeExport( '#', 'testing-2015-01-01.xml' ) );
-				//dispatch( failExport( 'The reason for failure would be displayed here' ) );
-			}, 1400 );
-		}, 400 );
+			// Poll for completion of the export
+			let poll = ( timeout ) => {
+				setTimeout( () => {
+					wpcomUndocumented.getExport( site.ID, 0, ( pollError, pollResponse ) => {
+						debug( pollError, pollResponse );
+						if ( pollError ) {
+							dispatch( failExport( ERROR_POLLING ) );
+							return;
+						}
+
+						if ( pollResponse.status === 'running' ) {
+							poll( 500 );
+						} else if ( pollResponse.status === 'finished' ) {
+							dispatch( completeExport( pollResponse.$attachment_url ) );
+						} else {
+							dispatch( failExport( ERROR_PROCESSING ) );
+						}
+					} );
+				}, timeout );
+			}
+			poll( 0 );
+		} );
```

Comment:
I think what's actually happening here is that our action creator ends up persisting in the background to interact with our app. That's somewhat confusing or complicated. What if instead of doing this here, your components were aware of the fact that the export had started and then displayed a progress component during that time. Simultaneously, if there an export were in progress, the React component could choose to poll for updating and choose how frequently to do it. By putting the polling here, it's hidden and less configurable.

When I made the importer, I created a special React component called `<Interval />` in `client/lib/interval` just for this purpose. Here's an example of how you could use it…

```js
import Interval from 'lib/interval';

import { pollExport } from './actions'

render() {
	const { exportIsRunning, exportIsFinished } = this.state;

	return (
		<div>
			…
			{ exportIsRunning && <Interval onTick={ pollExport } period={ EVERY_SECOND } /> }
			…
		</div>
	);
}
```